src/network/peloton_server.cpp

```diff
@@ -277,10 +278,9 @@ void PelotonServer::ServerLoop() {
   }
   dispatcher_task_->EventLoop();
   LOG_INFO("Closing server");
-  int status;
-  do {
-    status = close(listen_fd_);
-  } while (status < 0 && errno == EINTR);
+
+  peloton_close(listen_fd_);
+
```

Comment:
IMO the "Closing server" and "Server Closed" messages are bit too much now.
I would suggest removing the former ("Closing server") also.